seqr/management/commands/check_for_new_samples_from_pipeline.py

```
@@ -350,6 +359,59 @@ def _update_pdos(session, project_guid, sample_ids):
+        sample_individual_map = {
+            i.sample.sample_id: i for i in Individual.objects.filter(
+                family__project__in=projects,
+                sample__sample_id__in=sample_qc_map.keys(),
+            ).select_related('sample')
+        }
+
+        updated_individuals = []
+        unknown_filter_flags = set()
+        unknown_pop_filter_flags = set()
+
+        for sample_id, individual in sample_individual_map.items():
+            record = sample_qc_map[sample_id]
+            filter_flags = {}
+            for flag in json.loads(record['filter_flags']):
+                flag = '{}_{}'.format(flag, sample_type) if flag == 'coverage' else flag
+                flag_col = QC_FILTER_FLAG_COL_MAP.get(flag, flag)
+                if flag_col in record:
+                    filter_flags[flag] = record[flag_col]
+                else:
+                    unknown_filter_flags.add(flag)
+
+            pop_platform_filters = {}
+            for flag in json.loads(record['qc_metrics_filters']):
+                flag_col = 'sample_qc.{}'.format(flag)
+                if flag_col in record:
+                    pop_platform_filters[flag] = record[flag_col]
+                else:
+                    unknown_pop_filter_flags.add(flag)
+
+            individual.filter_flags.update(filter_flags)
+            individual.pop_platform_filters.update(pop_platform_filters)
+            individual.population.update(record['qc_pop'].upper())
+            updated_individuals.append(individual)
+
+        if updated_individuals:
+            Individual.objects.bulk_update(
+                updated_individuals,
+                ['filter_flags', 'pop_platform_filters', 'population'],
+            )
+
+        if unknown_filter_flags:
+            message = 'The following filter flags have no known corresponding value and were not saved: {}'.format(
+                ', '.join(unknown_filter_flags))
+            logger.warning(message)
```

warnings are useful when something is triggered in the UI and the person who does so will be shown the warnings, but they are essentially useless in a job that runs automatically in the background. 
I think we should hard fail updating QC if there are unexpected values, as we now have more control over how these are generated and can choose ot to include things we don't care about in our metadata

---

I agree here, this was a copy paste without critical thinking. There's actually no way we'd have erroneous filter flags, and for the hail qc_metrics_filters, which are set by hail's sample qc, we can trust that the output has the same flags every time, so I'm leaning towards just taking the flag validation out of the new seqr sample qc flow.